### PR TITLE
cdc, mvcc: always read old value when cdc requires (#10197)

### LINKED ISSUE #9981: CDC endpoint thread CPU usage 100% with insert 800ops
### What version of TiKV are you using?

tikv v4.0.6
ticdc v4.0.9

### Steps to reproduce

Creates a changefeed with old value, and executes 800ops insertion in upstream TiDB. 

### What did you expect?

CPU usage should be no more than 50%.

### LINKED ISSUE #9996: TiKV OOM due to CDC endpoint CPU 100% under 20k insert op/s
## Bug Report

image
image
image
image
image

From the above graph, OOM is caused by endpoint CPU usage is high, TiKV buffers too many scan events in memory.

### What version of TiKV are you using?

TiKV 5.0.0
TiCDC 5.0.0

### What operating system and CPU are you using?

TiKV: 8 CPU, 64GB memory

### What did you expect?

No TiKV OOM

### What did happened?

TiKV OOM

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #10197 to release-5.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/10246
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/10246:release-5.0-6c676b0af772
```

---

### What problem does this PR solve?

Issue Number:

Close #9996
Close #9981

Reading old value costs too much CPU for CDC endpoint thread, leading to OOM. 

### What is changed and how it works?

Always read old value in txn layer when cdc requires old value.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

Fix CDC OOM issue caused by reading old values.

### CODE REVIEW

File: src/storage/txn/actions/get_old_value.rs [hunk 1]
```diff
@@ -214,9 +220,16 @@ mod tests {
...
+                    .map(|w| w.1);
+                let prev_write_loaded = true;
+                let result = get_old_value(
+                    &mut txn,
+                    &Key::from_raw(b"a"),
+                    TimeStamp::new(25),
+                    prev_write_loaded,
+                    prev_write,
+                )
+                .unwrap();
```
Comment (MyonKeminta) [inline src/storage/txn/actions/get_old_value.rs:232]: There's a test case in master that checks prev_write_loaded is true and prev_write is none. It seems that case is missing here?

**Review by MyonKeminta — COMMENTED** [review]
Rest LGTM

**Review by MyonKeminta — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]

**Comment by overvenus** [comment]
/run-test

test_leader_demoted_when_hibernated  fails.

**Pull request merged**